sdk/servicebus/service-bus/src/core/shared.ts

```diff
@@ -0,0 +1,60 @@
+import { Delivery } from "rhea-promise";
+import { translate } from "@azure/core-amqp";
+import * as log from "../log";
+
+/**
+ * @internal
+ * @ignore
+ */
+export interface PromiseLike {
```

Comment:
Can we use a different name for this interface? There's already a `PromiseLike` in the typescript `lib.es5.d.ts` types that's meant to look like a Promise (versus this which does not match the promise API)

Reply:
Changed to `DeferredPromiseAndTimer`. Verbose, but at least it describes what it is. :)